test/e2e/bundle_test.go

```diff
@@ -18,6 +18,29 @@ import (
 	"github.com/vmware-tanzu/carvel-imgpkg/test/helpers"
 )
 
+func TestPushPullBundle(t *testing.T) {
+	env := helpers.BuildEnv(t)
+	imgpkg := helpers.Imgpkg{T: t, ImgpkgPath: env.ImgpkgPath}
+	defer env.Cleanup()
+
+	bundleDir := env.BundleFactory.CreateBundleDir(helpers.BundleYAML, helpers.ImagesYAML)
+
+	out := imgpkg.Run([]string{"push", "--tty", "-b", env.Image, "-f", bundleDir})
+	bundleDigest := fmt.Sprintf("@%s", helpers.ExtractDigest(t, out))
+
+	outDir := env.Assets.CreateTempFolder("imgpkg-test-basic")
+	imgpkg.Run([]string{"pull", "-b", env.Image + bundleDigest, "-o", outDir})
```

Comment:
I think in this case you want to pull by tag so you would have to pull using:
`env.Image:fmt.Sprintf("%s-%s.imgpkg", strings.Split(bundleDigest, ":")...)`